<?xml version="1.0"?>
<launch>
	<!-- ^^ old stuff ­­­| new stuff vv  -->

	<!-- Harware driver -->
	<node pkg="embla_hardware" type="embla_hardware_node" name="embla_hardware_node">
		<rosparam subst_value="true">
            control_frequency: 10.0
            diagnostic_frequency: 1.0

            max_acceleration: 2.0
            max_speed: 1.8
            wheel_diameter: 0.08
            pulses_per_rev: 990
            polling_timeout: 10.0
        </rosparam>
	</node>
	
	<!-- Load robot description -->
	<param name="robot_description" command="$(find xacro)/xacro '$(find embla_description)/urdf/embla.xacro' --inorder"/>
	
	<!-- Load controller configuration -->
	<rosparam command="load" file="$(find embla_hardware)/config/controllers.yaml"/>
	
	<!-- Publish static transforms from URDF and updated transforms from joint states -->
	<node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher"/>	<!-- state_publisher is the "old name" -->

	<!-- Spawn our own controllers. Names must match names in control config file -->
	<node name="base_controller_spawner" pkg="controller_manager" type="spawner" args="embla_joint_publisher embla_velocity_controller"/>
	

</launch>
